ext/checkers.jl

```
@@ -0,0 +1,230 @@
+                return false
+            end
+        else
+            # For the temporal dimension, only check if the times of metadata is
+            # a subset of the times of var, because _match_dates will get the
+            # correct dates for us
+            var_dates_or_times = dates_or_times(var)
+            metadata_dates_or_times = dates_or_times(metadata)
+            subset_of_var_dates = metadata_dates_or_times ⊆ var_dates_or_times
+            if !subset_of_var_dates
+                verbose &&
+                    @info "Dates/times of metadata ($metadata_dates_or_times) is not a subset of the dates/times of the OutputVar ($var_dates_or_times)"
+                return false
+            end
+        end
+    end
+    return true
+end
+
+"""
+    Checker.check(
+        ::SequentialIndicesChecker,
+        var::OutputVar,
+        metadata::Metadata;
+        verbose = false,
+    )
+
+Return `true` if the dates of `var` map to sequential indices of the dates of
+`metadata`, `false` otherwise.
+
+!!! note "Use this check"
+    It is recommended to always enable this check when possible.
+
+!!! note "Why use this check?"
+    This check is helpful in ensuring that the dates are matched correctly
+    between `var` and `metadata`. For example, without this check, if the
+    simulation data contain monthly averages and metadata track seasonal
+    averages, then no error is thrown, because all dates in `metadata` are in
+    all the dates in `var`.
+"""
+function Checker.check(
+    ::SequentialIndicesChecker,
+    var::OutputVar,
+    metadata::Metadata;
+    verbose = false,
+)
+    obs_dates = ClimaAnalysis.dates(metadata)
+    sim_dates = ClimaAnalysis.dates(var)
+    for dates in (obs_dates, sim_dates)
+        allunique(dates) || error("Dates in $dates are not unique")
```

Should this throw an error? I think this could just warn and return `false`.

---

I think it should throw an error, because there shouldn't be duplicate dates.

I can remove this though, since this is checked in `_match_dates`.